core/js/src/main/scala/org/http4s/js/crypto/Crypto.scala

```diff
@@ -0,0 +1,945 @@
+    js.Dynamic
+      .literal(name = name, prime = prime, generator = generator)
+      .asInstanceOf[DhKeyGenParams]
+}
+
+@js.native
+trait DhKeyAlgorithm extends KeyAlgorithm {
+  var prime: BigInteger = js.native
+
+  var generator: BigInteger = js.native
+}
+
+object DhKeyAlgorithm {
+  @inline
+  def apply(name: String, prime: BigInteger, generator: BigInteger): DhKeyAlgorithm =
+    js.Dynamic
+      .literal(name = name, prime = prime, generator = generator)
+      .asInstanceOf[DhKeyAlgorithm]
+}
+
+@js.native
+trait DhKeyDeriveParams extends Algorithm {
+  var `public`: CryptoKey = js.native
+}
+
+object DhKeyDeriveParams {
+  @inline
+  def apply(name: String, public: CryptoKey): DhKeyDeriveParams =
+    js.Dynamic
+      .literal(name = name, public = public)
+      .asInstanceOf[DhKeyDeriveParams]
+}
+
+@js.native
+trait DhImportKeyParams extends Algorithm {
+  var prime: BigInteger = js.native
+
+  var generator: BigInteger = js.native
+}
+
+object DhImportKeyParams {
+  @inline
+  def apply(name: String, prime: BigInteger, generator: BigInteger): DhImportKeyParams =
+    js.Dynamic
+      .literal(name = name, prime = prime, generator = generator)
+      .asInstanceOf[DhImportKeyParams]
+}
+
+// CONCAT
+
+@js.native
+trait ConcatParams extends Algorithm {
+  var hash: HashAlgorithmIdentifier = js.native
+
+  var algorithmId: BufferSource = js.native
+
+  var partyUInfo: BufferSource = js.native
+
+  var partyVInfo: BufferSource = js.native
+
+  var publicInfo: BufferSource = js.native
+
+  var privateInfo: BufferSource = js.native
+}
+
+object ConcatParams {
+  @inline
+  def apply(
+      name: String,
+      hash: HashAlgorithmIdentifier,
+      algorithmId: BufferSource,
+      partyUInfo: BufferSource,
+      partyVInfo: BufferSource,
+      publicInfo: BufferSource,
+      privateInfo: BufferSource): ConcatParams =
+    js.Dynamic
+      .literal(
+        name = name,
+        hash = hash.asInstanceOf[js.Any],
+        algorithmId = algorithmId,
+        partyUInfo = partyUInfo,
+        partyVInfo = partyVInfo,
+        publicInfo = publicInfo,
+        privateInfo = privateInfo
+      )
+      .asInstanceOf[ConcatParams]
+}
+
+// HKDF-CTR
+
+@js.native
+trait HkdfCtrParams extends Algorithm {
+  var hash: HashAlgorithmIdentifier = js.native
+
+  var label: BufferSource = js.native
+
+  var context: BufferSource = js.native
+}
+
+object HkdfCtrParams {
+  @inline
+  def apply(
+      name: String,
+      hash: HashAlgorithmIdentifier,
+      label: BufferSource,
+      context: BufferSource): HkdfCtrParams =
+    js.Dynamic
+      .literal(name = name, hash = hash.asInstanceOf[js.Any], label = label, context = context)
+      .asInstanceOf[HkdfCtrParams]
+}
+
+// PBKDF2
+
+@js.native
+trait Pbkdf2Params extends HashAlgorithm {
+  var salt: BufferSource = js.native
+
+  var iterations: Double = js.native
+
+  var hash: HashAlgorithmIdentifier = js.native
```

Comment:
Getting a ton of warnings like this.  Maybe we're stuck with them:

```scala
[warn] /Users/ross.baker/src/http4s/core/js/src/main/scala/org/http4s/js/crypto/Crypto.scala:863:42: dead code following this construct
[warn]   var hash: HashAlgorithmIdentifier = js.native
```

Reply:
I think 2.13 is better with this, because I tried `@nowarn`ing but then 2.13 started complaining there was nothing to `@nowarn`. This could probably be solved with a magic incantation of sbt settings.

Reply:
I'm not terribly worried about it, but there's enough scala.JS only code, it would be a nice mopup later so we don't miss bugs there by tolerating warnings.